src/LaunchDarkly.EventSource/EventSource.cs

```diff
@@ -252,12 +276,19 @@ private void ProcessResponseContent(string content)
 
         private void SetReadyState(ReadyState state, Action<StateChangedEventArgs> action = null)
         {
-            if (ReadyState == state) return;
-
-            ReadyState = state;
+            lock (this)
+            {
+                if (_readyState == state || _readyState == ReadyState.Shutdown)

```

Comment:
The second part of this test is important in order to avoid a race condition where `Close()` gets called, setting the state to Shutdown, and _then_ an error handler fires and sets it to Closed. Once it is Shutdown, it should stay that way.